src/ztex/device_format.c

```diff
@@ -79,15 +76,35 @@ void device_format_init(struct fmt_main *fmt_main,
 	jtr_devices_allow = devices_allow;
 	jtr_verbosity = verbosity;
 
-	struct list_entry *entry;
-	int found_bad_sn = 0;
-	for (entry = devices_allow->head; entry; entry = entry->next)
-		if (!ztex_sn_is_valid(entry->data)) {
-			fprintf(stderr, "Error: bad Serial Number '%s'\n", entry->data);
-			found_bad_sn = 1;
+
+	static int init_conf_devices = 0;
+	if (!init_conf_devices) {
+
+		// Initialize [List.ZTEX:Devices] configuration section.
+		ztex_sn_init_conf_devices();
+
+		// Check -dev command-line option.
+		struct list_entry *entry;
+		int found_error = 0;
+		for (entry = devices_allow->head; entry; entry = entry->next) {
+
+			if (isdigit(entry->data[0]) && (entry->data[1] == 0
+					|| (isdigit(entry->data[1]) && entry->data[2] == 0)) ) {
+				if (!ztex_sn_check_alias(entry->data))
+					found_error = 1;
+			}
+
+			else if (!ztex_sn_is_valid(entry->data)) {

```

Comment:
Coding style: I'd write this as `} else if (` on one line.

Reply:
I'll remove blank line (like ```indent -ce``` would do).
I'm not aware of coding style guidelines on the issue, and my preferece (as I explained in the above comment) is to do it base on a feeling if that's "too condensed" or not.
Sometimes I write ```} else if (```, ```} else {``` on one line.